tests/visualization_tests/test_pareto_front.py

```diff
@@ -123,41 +171,92 @@ def test_plot_pareto_front_3d(
     assert (figure.data[1]["z"] + figure.data[0]["z"]) == ()
 
     # Test with three trials.
+    study.enqueue_trial({"x": 1, "y": 1, "z": 2})
     study.enqueue_trial({"x": 1, "y": 1, "z": 1})
-    study.enqueue_trial({"x": 1, "y": 0, "z": 1})
+    study.enqueue_trial({"x": 1, "y": 0, "z": 2})
     study.enqueue_trial({"x": 1, "y": 1, "z": 0})
     study.optimize(
-        lambda t: [t.suggest_int("x", 0, 1), t.suggest_int("y", 0, 1), t.suggest_int("z", 0, 1)],
-        n_trials=3,
+        lambda t: [t.suggest_int("x", 0, 1), t.suggest_int("y", 0, 2), t.suggest_int("z", 0, 2)],
+        n_trials=4,
     )
 
+    constraints_func: Optional[Callable[[FrozenTrial], Sequence[float]]]
+    if use_constraints_func:
+        # (x, y, z) = (1, 0, 1) is infeasible; others are feasible.
```

Comment:
The infeasible point seems to be (1, 1, 0).
```suggestion
        # (x, y, z) = (1, 1, 0) is infeasible; others are feasible.
```